include/Gaffer/MetadataAlgo.h

```diff
@@ -91,6 +91,9 @@ GAFFER_API bool getReadOnly( const GraphComponent *graphComponent );
 GAFFER_API void setChildNodesAreReadOnly( Node *node, bool readOnly, bool persistent = true );
 GAFFER_API bool getChildNodesAreReadOnly( const Node *node );
 
+/// Looks for the most distant parent of node whose children are all read-only.
+GAFFER_API Node *ancestorWithReadOnlyChildNodes( Node *node );
```

Comment:
Do you think there's a better way of phrasing this? One of the reasons I kept it private originally was that it sounded so awkward, and didn't even quite do what I wanted anyway. At both call sites we just want to know which node to show, `node` or some ancestor, but this doesn't do that, because it sometimes returns `null`. Perhaps `closestEditableAncestor()`?

Reply:
As per out-of-band discussion. We're confining this to private implementation for now, until we can figure out a sane way to adopt this in `MetadataAlgo`.